src/preact/context.js

```diff
@@ -77,3 +85,35 @@ export function useAmpContext() {
   const AmpContext = getAmpContext();
   return useContext(AmpContext);
 }
+
+/**
+ * Whether the calling component should currently be in the loaded state.
+ *
+ * @param {!Loading|string} loadingProp
+ * @return {boolean}
+ */
+export function useLoad(loadingProp) {
+  const loadingRef = useRef(false);
+
+  const {loading: loadingContext, renderable} = useAmpContext();
+
+  const loading = loadingReducer(loadingProp, loadingContext);
+
+  // Compute whether the element should be loading at this time.
+  const load =
+    // Explicit instruction to unload.
+    loading == Loading.UNLOAD
+      ? false
+      : // Explicit instrunction to load.
+      loading == Loading.EAGER
+      ? true
+      : // Auto: allowed to load.
+      loading == Loading.AUTO && renderable
+      ? true
+      : // Lazy: can continue loading if already started, but do not start it.
+        loadingRef.current || false;
+  if (loadingRef.current != load) {
```

Comment:
nit: can nix the if condition unless you think the conditional assignment buys us something

Reply:
Nope. Nothing. I think I just used it for debugging and it stuck.